qiskit_ibm_provider/utils/json_decoder.py

```diff
@@ -69,6 +83,152 @@ def properties_from_server_data(properties: Dict) -> BackendProperties:
+            )
+        else:
+            warnings.warn(
+                f"Definition of instruction {name} is not found in Qiskit and "
+                "no gate configuration is provided by the backend. "
+                "This instruction is not created in Target.",
+                UserWarning,
+            )
+            all_instructions.remove(name)
+            continue
+
+    # Create placeholder for the instruction properties
+    for name, spec in gate_configs.items():
+        if hasattr(spec, "coupling_map"):
+            coupling_map = spec.coupling_map
+            prop_name_map[name] = dict.fromkeys(map(tuple, coupling_map))
+        else:
+            prop_name_map[name] = None
+    if "delay" not in prop_name_map:
+        # Case for real IBM backend. They don't have delay in gate configuration.
+        prop_name_map["delay"] = {(q,): None for q in range(configuration.n_qubits)}
+
+    # Populate instruction properties
+    if properties:
+        in_data["qubit_properties"] = list(
+            map(_decode_qubit_property, properties["qubits"])
+        )
+        for gate_spec in map(GateSchema.from_dict, properties["gates"]):
+            inst_prop = _decode_instruction_property(gate_spec)
+            qubits = tuple(gate_spec.qubits)
+            name = gate_spec.gate
+            if name not in all_instructions:
+                # New instruction is found in property. Likely an edge case.
+                new_instruction = Gate(
+                    name=name,
+                    num_qubits=len(qubits),
+                    params=[],
+                )
+                prop_name_map[name] = {}
+                inst_name_map[name] = new_instruction
+                all_instructions.add(name)
+            if prop_name_map[name] is None:
+                prop_name_map[name] = {}
+            prop_name_map[name][qubits] = inst_prop
+        # Measure instruction property is stored in qubit property in IBM
+        measure_props = list(map(_decode_measure_property, properties["qubits"]))
+        prop_name_map["measure"] = {}
+        for qubit, measure_prop in enumerate(measure_props):
+            qubits = (qubit,)
+            prop_name_map["measure"][qubits] = measure_prop
+
+    # Define pulse qobj converter and command sequence for lazy conversion
+    if pulse_defaults:
+        pulse_lib = list(
+            map(PulseLibraryItem.from_dict, pulse_defaults["pulse_library"])
+        )
+        converter = QobjToInstructionConverter(pulse_lib)
+        for cmd in map(Command.from_dict, pulse_defaults["cmd_def"]):
+            name = cmd.name
+            qubits = tuple(cmd.qubits)
+            if name not in all_instructions or qubits not in prop_name_map[name]:
+                # Ignore pulse gate.
+                continue
+            entry = PulseQobjDef(converter=converter, name=cmd.name)
+            entry.define(cmd.sequence)
+            prop_name_map[name][qubits].calibration = entry
```

Comment:
Did you want to have any error handling for the case something was in the pulse defaults and not in configuration or properties? I don't think we should add an opaque gate in this case, but may just a more descriptive user error. Something like:

```suggestion
            try:
                prop_name_map[name][qubits].calibration = entry
            except AttributeError as e:
                raise IBMBackendApiProtocolError(
                    f"The PulseDefaults payload received contains an instruction {name} on "
                    f"qubits {qubits} which is not present in the configuration or properties "
                    "payload
                ) from e
```

I only mention this because during a recent new device bring-up this exact error occurred and others had trouble debugging it until I pointed out what was wrong.